### ISSUE TITLE
FNX-12223 ⁃ [Bug] Tabtray bottom sheet needs changes to work properly with top toolbar layout

### ISSUE THREAD

**Issue by DavidBE-UK** [mozilla-mobile/fenix#11314]
N.B. This is a BUG, not a feature request, as the current tabtray implementation has clearly been designed purely with the toolbar bottom layout in mind, and no thought to  how it should work with the toolbar top layout. That is a fundamental design flaw, not just missing a few extra bells and whistles someone might want. If there is a main layout top toolbar option, then the rest of the UI should allow for that with appropriate changes kicking in when toolbar top is set. Otherwise you might as well just scrap the toolbar top option and people like me who require tools at the top will just go to another browser.

## Steps to reproduce
With Toolbar position set to TOP, press tab switcher button.

### Expected behavior
When the display configuration is set to toolbar TOP, whatever opens should list open tabs from the TOP down, and there should be a control at/near the top to open a new tab. This could either be:

1. Tabtray bottom sheet as it is now but when main display option toobar top is set then its behaviour is modified so that it opens the tabtray right the way up (as it can currently be dragged up to), and either some button added at the top or an entry in its three dot menu to open a new tab;

or

1. An option to use a conventional tab switcher page instead of the bottom sheet tabtray, that page showing open tabs working down for the top and a button to open a new tab at the top as in current Fennic AND Android Chrome AND Android Edge.

### Actual behavior
Bottom sheet opens but only comes up part way, so I have to uncomfortably stretch thumb down and actually usually have to shift my whole hand down to either tap on the first few tabs or hold-drag the top of the tabtray bottom sheet up to its upper position so the tabs listed go from the top down just below the toolbar as they should for someone using top toolbar position. And the New Tab action button is WAY out of reach at the bottom.

It is entirely designed to go with toolbar bottom layout, not top. Its behaviour should change as indicated above (opens right the way up, and  with a new tab button or entry in the 3 dot menu) when toolbar top configuration is set.

I liked top controls even before I became disabled. Now I require them. Since my last 'must have' appeared - being able to use dark reader - I have been using Fenix Nightly as my default browser on my phone. Now, with this, I am not. Providing a top toolbar display option requires considering and allowing for it _and the reason it is wanted_ with all the other aspects of the UI. The current tabtray bottom sheet behaviour when top toolbar is set (i.e. same as when bottom toolbar is set) is NOT allowing for it. With it as it is I will not be using Nightly as my browser. If it stays like this with no allowance for top toolbar users and that goes on into the beta, I will be moving to another browser on my phone, and the need for syncing between phone and laptop means that will mean me giving up desktop Firefox too. I say this just to make clear that this is a make-or-break issue for me. If you are someone who likes bottom controls you may not get it, but this is that serious an issue for me.

### Device information
* Android device: ? Alcatel3V TCL 5099Y
* Fenix version: ? Nightly 200606 06:21 (Build #21580627)

**Comment by ekager** [comment]
Please do not spam our issue list with dupes just because you don't like the labels added. You can add more information in #11314 that we can get to when we have time to respond to your concerns.

**Comment by DavidBE-UK** [comment]
I submitted a new issue because it is for something different. In the previous issue I asked for a tab tray sheet coming down from the top, and was told this is an Android bottom sheet element that cannot do that. I took that as being the end of that issue, and went and read about Android Bottom sheets and realised my previous issue assumed things that weren't true or possible. So I then considered the problem and this is a reconsidered report based on what actually appears to be possible. Sorry if that offends - it was an attempt at enhancing clarity with a new issue that could be progressed whereas I presumed my impossible old issue would be closed. This was not spamming.
